sdk/src/trace/span.h

```diff
@@ -22,7 +22,9 @@ class Span final : public trace_api::Span
                 const trace_api::SpanContextKeyValueIterable &links,
                 const trace_api::StartSpanOptions &options,
                 const trace_api::SpanContext &parent_span_context,
-                const opentelemetry::sdk::resource::Resource &resource) noexcept;
+                const opentelemetry::sdk::resource::Resource &resource,
+                const nostd::shared_ptr<opentelemetry::trace::TraceState> trace_state =
```

Comment:
FYI, this might be conflicted with the `trace_flags` change which needs to be resolved.